src/Illuminate/Database/Connectors/Connector.php

```diff
@@ -63,10 +62,6 @@ public function createConnection($dsn, array $config, array $options)
      */
     protected function createPdoConnection($dsn, $username, $password, $options)
     {
-        if (class_exists(PDOConnection::class) && ! $this->isPersistentConnection($options)) {
-            return new PDOConnection($dsn, $username, $password, $options);
-        }
-
```

Comment:
This was the weirdest quirk I found. This part is checking for `Doctrine\DBAL\Driver\PDOConnection` which doesn't exists anymore in Doctrine DBAL v3 (it's `Doctrine\DBAL\Driver\PDO\Connection`). So this has never worked when people installed Doctrine DBAL v3 (on Laravel v9). Given that we never have gotten reports about it I feel like it's safe to remove this part.